File: integration_tests/connector/go/ConnectTest.go
package main

import (
	"crypto/tls"
	"database/sql"
	"fmt"
	"github.com/go-sql-driver/mysql"
	"log"
	"os"
)

func main() {

	host := os.Args[1]
	user := os.Args[2]
	password := os.Args[3]

	mysql.RegisterTLSConfig("tidb", &tls.Config{
		MinVersion: tls.VersionTLS12,
		ServerName: host,
	})

	db, err := sql.Open("mysql", fmt.Sprintf("%s:%s@tcp(%s:4000)/test?tls=tidb", user, password, host))
	if err != nil {
		log.Fatal("failed to connect database", err)
	}
	defer db.Close()

	var dbName string
	err = db.QueryRow("SELECT DATABASE();").Scan(&dbName)
	if err != nil {
		log.Fatal("failed to execute query", err)
	}
	fmt.Println(dbName)
}

File: integration_tests/connector/node/ConnectTest.js
var mysql = require('mysql2');



var connection = mysql.createConnection({
  host:  process.argv[2],
  port: 4000,
  user: process.argv[3],
  password: process.argv[4],
  database: 'test',
  ssl: {
    minVersion: 'TLSv1.2',
    rejectUnauthorized: true
  }
});
connection.connect(function(err) {
  if (err) {
    throw err
  }
  connection.query('SELECT DATABASE();', function(err, rows) {
    if (err) {
      throw err
    }
    console.log(rows[0]['DATABASE()']);
    connection.end()
  });
});